### PR TITLE
Add start, stop

### PR DESCRIPTION
**Pull request by d2lam**
Add start and stop functions for base class

### CODE REVIEW

**Comment by nkatzman** [comment]
👍

File: index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+     */
+    constructor(config) {
+        this.config = config;
+    }
+
+    /**
+     * Start the executor
+     * @method start
+     */
+    start() {
```
Comment (stjohnjohnson) [inline index.js]: What fields?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        console.error('start is not implemented');
```
Comment (stjohnjohnson) [inline index.js]: Don't use console, use winston.

Comment (nkatzman) [inline index.js]: well the reason petey used console in the datastore-base was that he wanted the base class as small as possible and not bring in any external dependencies

File: index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+/* eslint-disable no-console */
+
+class Executor {
```
Comment (stjohnjohnson) [inline index.js]: This class should be defining the interface for executors, I don't see that.

**Comment by stjohnjohnson** [comment]
Why is the readme not updated?

**Comment by d2lam** [comment]
@stjohnjohnson  Added README

File: index.js (hunk 3 above)
Comment (d2lam) [inline index.js]: I added the fields to the comments. We can add validation during hip week for this and also the datastore-base?

**Comment by nkatzman** [comment]
Seems reasonable 👍

**Comment by stjohnjohnson** [comment]
👍

**Pull request merged**